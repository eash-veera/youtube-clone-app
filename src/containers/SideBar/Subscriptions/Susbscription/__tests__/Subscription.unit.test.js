import React from 'react';
import { shallow } from 'enzyme';
import Subscription from '../Subscription';

describe('Subscription', () => {

    test('Renders empty subscription', () => {
        const wrapper = shallow(
            <Subscription />
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('Renders breadcasting subscription', () => {
        const wrapper = shallow(
            <Subscription broadcasting label='Kanywest'/>
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('Renders non-breadcasting subscription with new videos', () => {
        const wrapper = shallow(
            <Subscription label='Halsey Music' amountNewVideos={8}/>
        );
        expect(wrapper).toMatchSnapshot();
    });

});

