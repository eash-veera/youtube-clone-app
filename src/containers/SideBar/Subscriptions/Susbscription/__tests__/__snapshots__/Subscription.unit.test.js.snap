// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subscription Renders breadcasting subscription 1`] = `
<MenuItem>
  <div
    className="subscription"
  >
    <div>
      <Image
        as="img"
        avatar={true}
        src="http://via.placeholder.com/28x28"
        ui={true}
      />
      <span>
        Kanywest
      </span>
    </div>
    <Icon
      as="i"
      name="signal"
    />
  </div>
</MenuItem>
`;

exports[`Subscription Renders empty subscription 1`] = `
<MenuItem>
  <div
    className="subscription"
  >
    <div>
      <Image
        as="img"
        avatar={true}
        src="http://via.placeholder.com/28x28"
        ui={true}
      />
      <span />
    </div>
  </div>
</MenuItem>
`;

exports[`Subscription Renders non-breadcasting subscription with new videos 1`] = `
<MenuItem>
  <div
    className="subscription"
  >
    <div>
      <Image
        as="img"
        avatar={true}
        src="http://via.placeholder.com/28x28"
        ui={true}
      />
      <span>
        Halsey Music
      </span>
    </div>
    <span
      className="new-videos-count"
    >
      8
    </span>
  </div>
</MenuItem>
`;
